### PR TITLE
Adding new validating OSD LVM metadata section (BSC#1152333)

### PR DESCRIPTION
**Pull request by asettle**

### CODE REVIEW

File: xml/admin_install_salt.xml [hunk 1]
```diff
@@ -684,6 +684,12 @@ done</screen>
   skip=$((`blockdev --getsz /dev/sdX` - 33)) | hexdump -C
 </screen>
      </step>
+     <step>
+       <para>Ensure you reboot the OSDs after running the <literal>dd</literal>
+       command, otherwise you risk OSD data loss.
+       Run the following command to reboot:</para>
+<screen>&prompt.root; reboot</screen>
```
Comment (tbazant) [inline xml/admin_install_salt.xml]:
```suggestion
<screen>&prompt.root;reboot</screen>
```

**Comment by asettle** [comment]
Blocking merge on this until Boyd is able to provide context in the bug report :) thanks

**Comment by asettle** [comment]
Based on Boyd's comment here: https://bugzilla.suse.com/show_bug.cgi?id=1152333#c6
This PR will be completely revised and a second commit will come shortly.

File: xml/admin_saltcluster.xml [hunk 2]
```diff
@@ -546,6 +546,92 @@ True
...
+  devices /dev/sda
+</screen>
+      <para>In this example, you can see that <literal>osd.2</literal> is still located in
+      <filename>/dev/sda</filename>.</para>
+      </step>
+      <step>
+        <para>
+          On the OSD node, validate the LVM metadata:
+        </para>
+<screen>&prompt.smaster; ceph-volume inventory</screen>
```
Comment (tbazant) [inline xml/admin_saltcluster.xml]:
```suggestion
<screen>&prompt.smaster;ceph-volume inventory</screen>
```

File: xml/admin_saltcluster.xml [hunk 3]
```diff
@@ -546,6 +546,92 @@ True
...
+  /dev/sdf 40.00 GB True False QEMU HARDDISK
+  /dev/vda 25.00 GB True False
+</screen>
+      </step>
+      <step>
+        <para>
+          Run the following command on the OSd node to remove the LVM metadata
+          completely:
+        </para>
+<screen>&prompt.smaster; ceph-volume lvm zap --osd-id <replaceable>ID</replaceable> --destroy </screen>
```
Comment (tbazant) [inline xml/admin_saltcluster.xml]:
```suggestion
<screen>&prompt.smaster;ceph-volume lvm zap --osd-id <replaceable>ID</replaceable> --destroy </screen>
```

File: xml/admin_saltcluster.xml [hunk 4]
```diff
@@ -546,6 +546,92 @@ True
...
+      </step>
+      <step>
+        <para>
+          Run the <command>inventory</command> again to verify that the <filename>/dev/sda</filename>
```
Comment (tbazant) [inline xml/admin_saltcluster.xml]:
```suggestion
          Run the <command>inventory</command> command again to verify that the <filename>/dev/sda</filename>
```

File: xml/admin_saltcluster.xml [hunk 5]
```diff
@@ -546,6 +546,92 @@ True
...
+          availability returns <literal>True</literal>. For example:
+        </para>
+<screen>&prompt.smaster; ceph-volume inventory
```
Comment (tbazant) [inline xml/admin_saltcluster.xml]:
```suggestion
<screen>&prompt.smaster;ceph-volume inventory
```

File: xml/admin_saltcluster.xml [hunk 6]
```diff
@@ -546,6 +546,92 @@ True
...
+/dev/sde 40.00 GB True False QEMU HARDDISK
+/dev/sdf 40.00 GB True False QEMU HARDDISK
+/dev/vda 25.00 GB True False</screen>
+        <para>LVM metadata is now removed. It is now safe to <command>dd</command> the device.</para>
+      </step>
+      <step>
+        <para>
+          The OSD can now be re-deployed without needing to reboot the OSD node:
+        </para>
+<screen>&prompt.smaster; salt-run state.orch ceph.stage.2
```
Comment (tbazant) [inline xml/admin_saltcluster.xml]:
```suggestion
<screen>&prompt.smaster;salt-run state.orch ceph.stage.2
```

File: xml/admin_saltcluster.xml [hunk 7]
```diff
@@ -546,6 +546,92 @@ True
...
+&prompt.smaster; salt-run state.orch ceph.stage.3
```
Comment (tbazant) [inline xml/admin_saltcluster.xml]:
```suggestion
&prompt.smaster;salt-run state.orch ceph.stage.3
```

File: xml/admin_saltcluster.xml [hunk 8]
```diff
@@ -546,6 +546,93 @@ True
     </para>
    </tip>
   </sect2>
+  <sect2 xml:id="validate-osd-lvm">
+    <title>Validating OSD LVM metadata</title>
```
Comment (tbazant) [inline xml/admin_saltcluster.xml]:
```suggestion
    <title>Validating OSD LVM Metadata</title>
```

File: xml/admin_saltcluster.xml [hunk 9]
```diff
@@ -546,6 +546,93 @@ True
...
+  /dev/sdc 40.00 GB True False QEMU HARDDISK
+  /dev/sdd 40.00 GB True False QEMU HARDDISK
+  /dev/sde 40.00 GB True False QEMU HARDDISK
+  /dev/sdf 40.00 GB True False QEMU HARDDISK
+  /dev/vda 25.00 GB True False
+</screen>
+      </step>
+      <step>
+        <para>
+          Run the following command on the OSd node to remove the LVM metadata
```
Comment (jan--f) [inline xml/admin_saltcluster.xml]:
```suggestion
          Run the following command on the OSD node to remove the LVM metadata
```

File: xml/admin_saltcluster.xml [hunk 10]
```diff
@@ -546,6 +546,93 @@ True
     </para>
    </tip>
   </sect2>
+  <sect2 xml:id="validate-osd-lvm">
+    <title>Validating OSD LVM Metadata</title>
+    <para>After removing the OSD, sometimes the LVM metadata is not completely
```
Comment (jan--f) [inline xml/admin_saltcluster.xml]:
```suggestion
    <para>After removing an OSD via salt-run osd.remove <id> or through ceph commands, LVM metadata may not be completely
```

**Comment by asettle** [comment]
> Spotted a typo.
> 
> The other suggestions mostly aims at adding some context. There could probably be a link to the "remove OSD" doc if there is.

The doc is within this same section :)

**Pull request merged**